src/components/header/index.tsx

```diff
@@ -27,9 +49,20 @@ const Header = (props: HeaderProps) => {
         top: 0,
       }}
     >
+      {/* TODO: similarly refactor generic info alert messages into a centralized util file
+        and corresponding piece of global state; use this to set the `top` attribute passed to error alerts below.
+      */}
       {isAppRunningInStaging() ? <StagingBanner /> : <BetaBanner />}
-      <FetchError />
-      <InvalidJsonErrorBanner />
+
+      {errorTypesToDisplay.map((error: [string, string], i: number) => {
+        return (
+          <GenericErrorBanner
+            errorCategory={error[0]}
```

Comment:
This will work, but I find it more readable to have your error be an object rather than an array. Then you access by attribute/key rather than index. It's more readable to have some sort of `Error` interface that always has an `error.category` and `error.type` guaranteed on it, then when you extract those here it's more clear what's going on. (Also, just generally, I think access by index is a lot more prone to mistakes / awful debugging experiences than named keys/attrs. You can describe intent with the latter, but the former requires awareness of the structure from reading documentation/code elsewhere.)